src/python/preprocess.rs

```diff
@@ -0,0 +1,296 @@
+use crate::color::DicomColorType;
+use crate::load::LoadFromTiff;
+use crate::preprocess::Preprocessor;
+use crate::python::path::PyPath;
+use crate::save::TiffSaver;
+use crate::transform::resize::FilterType;
+use crate::transform::volume::{CentralSlice, KeepVolume, VolumeHandler};
+use crate::transform::PaddingDirection;
+use ::tiff::decoder::Decoder;
+use dicom::object::{from_reader, open_file, FileDicomObject, InMemDicomObject};
+use ndarray::Array4;
+use num::Zero;
+use numpy::Element;
+use numpy::{IntoPyArray, PyArray4};
+use pyo3::buffer::PyBuffer;
+use pyo3::prelude::*;
+use pyo3::{
+    exceptions::{PyFileNotFoundError, PyRuntimeError},
+    pymodule,
+    types::{PyAnyMethods, PyModule},
+    Bound, PyAny, PyResult, Python,
+};
+use std::clone::Clone;
+use std::io::Seek;
+use std::path::Path;
+use tempfile::spooled_tempfile;
+use tiff::encoder::compression::{Compressor, Uncompressed};
+use tiff::encoder::TiffEncoder;
+
+const SPOOL_SIZE: usize = 1024 * 1024 * 64;
```

Comment:
I think a comment & justification (high-level) for this spool size would be helpful.

Reply:
Further explanation added to README